chromium_src/v8/src/inspector/value-mirror.cc

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "src/v8/src/inspector/value-mirror.cc"
+
+#include "brave/components/brave_page_graph/common/buildflags.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_PAGE_GRAPH)
+
+#include "brave/v8/include/v8-isolate-page-graph-utils.h"
+#include "v8-local-handle.h"
+#include "v8-primitive.h"
+
+namespace v8::page_graph {
+
+namespace {
+
+class PropertyMirrors : public v8_inspector::ValueMirror::PropertyAccumulator {
+ public:
+  PropertyMirrors() {}
+  bool Add(v8_inspector::PropertyMirror mirror) override {
+    m_mirrors_.push_back(std::move(mirror));
+    return true;
+  }
+
+  auto& mirrors() { return m_mirrors_; }
+
+ private:
+  std::vector<v8_inspector::PropertyMirror> m_mirrors_;
+};
+
+// Returns true if the value is an empty object.
```

Comment:
Frankly, there are some other places which deserve a comment much more than this function :D